openslides_backend/shared/otel.py

```diff
@@ -0,0 +1,56 @@
+import os
+from contextlib import nullcontext
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+
+from .env import is_truthy
+
+"""
+Initializes the opentelemetry components and connection to the otel collector.
+"""
+def init(service_name):
+    if not is_truthy(os.environ.get("OPENTELEMETRY_ENABLED", "false")):
+        return
+
+    span_exporter = OTLPSpanExporter(
+        endpoint=f"http://collector:4317",
```

Comment:
Obsolete if you import this form the datastore, but this should probably also be configurable via env var

Reply:
There already is an environment variable. See https://opentelemetry-python.readthedocs.io/en/latest/exporter/otlp/otlp.html

Since OpenTelementry is build for micro-services, they made a standard for environment variables. With this variables, you do not only configure your own services, but also other servies like caddy or redis (if they add support for OpenTelementry in the future).

I would recommend, that we use this environment variables instead of inventing our own.

Reply:
I agree

Reply:
Can you change this to use the OTEL env vars?